backend/src/main/java/org/cryptomator/hub/api/DeviceResource.java

```diff
@@ -174,10 +175,18 @@ public record DeviceDto(@JsonProperty("id") @ValidId String id,
 							@JsonProperty("publicKey") @NotNull @OnlyBase64Chars String publicKey,
 							@JsonProperty("userPrivateKey") @NotNull @ValidJWE String userPrivateKeys, // singular name for history reasons (don't break client compatibility)
 							@JsonProperty("owner") @ValidId String ownerId,
-							@JsonProperty("creationTime") Instant creationTime) {
+							@JsonProperty("creationTime") Instant creationTime,
+							@JsonProperty("lastIpAddress") String lastIpAddress,
+							@JsonProperty("lastAccessTime") Instant lastAccessTime) {
 
 		public static DeviceDto fromEntity(Device entity) {
-			return new DeviceDto(entity.getId(), entity.getName(), entity.getType(), entity.getPublickey(), entity.getUserPrivateKeys(), entity.getOwner().getId(), entity.getCreationTime().truncatedTo(ChronoUnit.MILLIS));
+			return new DeviceDto(entity.getId(), entity.getName(), entity.getType(), entity.getPublickey(), entity.getUserPrivateKeys(), entity.getOwner().getId(), entity.getCreationTime().truncatedTo(ChronoUnit.MILLIS), null, null);
+		}
+
+		public static DeviceDto fromEntity(Device d, VaultKeyRetrievedEvent event) {
```

Comment:
As it is rather uncommon that we explicitly allow null, I'd mark this parameter accordingly (until [we get to enjoy this on a language level](https://openjdk.org/jeps/8303099)):

```suggestion
		public static DeviceDto fromEntity(Device d, @Nullable VaultKeyRetrievedEvent event) {
```

Reply:
Applied in [8436980](https://github.com/cryptomator/hub/pull/320/commits/84369806069a678a8ffdbb7878291288215f9e76)